src/hu/elte/txtuml/stdlib/net/URIImpl.java

```diff
@@ -0,0 +1,625 @@
+     *
+     * <p> This method is provided for use in situations where it is known that
+     * the given string is a legal URI, for example for URI constants declared
+     * within in a program, and so it would be considered a programming error
+     * for the string not to parse as such.  The constructors, which throw
+     * {@link URISyntaxException} directly, should be used situations where a
+     * URI is being constructed from user input or from some other source that
+     * may be prone to errors.  </p>
+     *
+     * @param  str   The string to be parsed into a URI
+     * @return The new URI
+     *
+     * @throws  NullPointerException
+     *          If {@code str} is {@code null}
+     *
+     * @throws  IllegalArgumentException
+     *          If the given string violates RFC&nbsp;2396
+     */
+    public static URI create(String str) {
+        try {
+            return new URIImpl(str);
+        } catch (URISyntaxException x) {
+            throw new IllegalArgumentException(x.getMessage(), x);
+        }
+    }
+    
+    /**
+     * Tests this URI for equality with another object.
+     *
+     * <p> If the given object is not a URI then this method immediately
+     * returns {@code false}.
+     *
+     * <p> For two URIs to be considered equal requires that either both are
+     * opaque or both are hierarchical.  Their schemes must either both be
+     * undefined or else be equal without regard to case. Their fragments
+     * must either both be undefined or else be equal.
+     *
+     * <p> For two opaque URIs to be considered equal, their scheme-specific
+     * parts must be equal.
+     *
+     * <p> For two hierarchical URIs to be considered equal, their paths must
+     * be equal and their queries must either both be undefined or else be
+     * equal.  Their authorities must either both be undefined, or both be
+     * registry-based, or both be server-based.  If their authorities are
+     * defined and are registry-based, then they must be equal.  If their
+     * authorities are defined and are server-based, then their hosts must be
+     * equal without regard to case, their port numbers must be equal, and
+     * their user-information components must be equal.
+     *
+     * <p> When testing the user-information, path, query, fragment, authority,
+     * or scheme-specific parts of two URIs for equality, the raw forms rather
+     * than the encoded forms of these components are compared and the
+     * hexadecimal digits of escaped octets are compared without regard to
+     * case.
+     *
+     * <p> This method satisfies the general contract of the {@link
+     * java.lang.Object#equals(Object) Object.equals} method. </p>
+     *
+     * @param   ob   The object to which this object is to be compared
+     *
+     * @return  {@code true} if, and only if, the given object is a URI that
+     *          is identical to this URI
+     */
+    public boolean equals(Object ob) {
```

Comment:
`public boolean equals(Object ob)` is a method of `java.lang.Object` which I override to make comparison work properly.  It is not logical part of `URI` interface, because of that I intentionally didn't put it there. Even if we add it to the interface, the compiler will not complain as it already inherits `equals(Object ob)` from the base `java.lang.Object` class.

It is important that `equals` takes an `Object`. Otherwise, unexpected behavior occurs. Consider the following scenario if we take `URI` as an argument instead of `Object`.
```
URI uriInstance1 = new URI("http://google.com");
URI uriInstance2 = new URI("http://google.com");
Object objInstance = new URI("http://google.com");

uriInstance1.equals(uriInstance2); //returns true
uriInstance1.equals(objInstance); // returns false, because URI.equals(Object) is inherited from java.lang.Object 
```

In addition, since `java.lang.Object` is a superclass of the all Java classes, in my opinion, it should not make any problem. Please let me know if you agree.